apollo-router/src/apollo_studio_interop/mod.rs

```diff
@@ -356,33 +405,60 @@ fn format_selection_set(selection_set: &SelectionSet, f: &mut fmt::Formatter) ->
     }
 
     if !fields.is_empty() || !named_fragments.is_empty() || !inline_fragments.is_empty() {
-        fields.sort_by(|&a, &b| a.name.cmp(&b.name));
+        if is_enhanced(normalization_algorithm) {
+            // in enhanced mode we display aliases so we show non-aliased field sorted by name first, then aliased fields sorted by alias
+            fields.sort_by(|&a, &b| {
+                match (a.alias.as_ref(), b.alias.as_ref()) {
+                    (None, None) => a.name.cmp(&b.name), // when both are non-aliased, sort by field name
+                    (Some(alias_a), Some(alias_b)) => alias_a.cmp(alias_b), // when both are aliased, sort by alias
+                    // when one is aliased and on isn't, the non-aliased field comes first
+                    (Some(_), None) => Ordering::Greater,
+                    (None, Some(_)) => Ordering::Less,
+                }
+            });
+        } else {
+            // otherwise we just sort by field name (and remove aliases in the field)
+            fields.sort_by(|&a, &b| a.name.cmp(&b.name));
+        }
+
+        // named fragments are always sorted
         named_fragments.sort_by(|&a, &b| a.fragment_name.cmp(&b.fragment_name));
-        // Note that inline fragments are not sorted in the JS implementation
+
+        // in enhanced mode we sort inline fragments
+        if is_enhanced(normalization_algorithm) {
+            inline_fragments.sort_by(|&a, &b| {
+                let a_name = a.type_condition.as_ref().map(|t| t.as_str()).unwrap_or("");
+                let b_name = b.type_condition.as_ref().map(|t| t.as_str()).unwrap_or("");
+                a_name.cmp(b_name)
+            });
+        }
 
         f.write_str("{")?;
 
         for (i, &field) in fields.iter().enumerate() {
-            let field_str = ApolloReportingSignatureFormatter::Field(field).to_string();
+            let formatter = SignatureFormatterWithAlgorithm {
+                formatter: &ApolloReportingSignatureFormatter::Field(field),
+                normalization_algorithm,
+            };
+            let field_str = format!("{}", formatter);
             f.write_str(&field_str)?;
```

Comment:
```suggestion
            write!(f, "{formatter}")?;
```
To remove the intermediate string allocation

Reply:
For this one I actually need the intermediate string because the behaviour after this call depends on the last character of field_str.